### ISSUE TITLE
Anonsurf Problems Kali Linux, not working

### ISSUE THREAD

**Issue by Resire** [Und3rf10w/kali-anonsurf#16]
OK I have tried everything to make this work. I had Kali Linux duel booted with Windows 10 and it did not work first of all 2nd I even reinstalled Kali Linux on my other HDD which had nothing on it and then tried anonsurf again and it didnt work I have looked everywhere and really need help why this is not working.

root@kali:# git clone https://github.com/Und3rf10w/kali-anonsurf.git
Cloning into 'kali-anonsurf'...
remote: Counting objects: 301, done.
remote: Compressing objects: 100% (23/23), done.
remote: Total 301 (delta 5), reused 0 (delta 0), pack-reused 278
Receiving objects: 100% (301/301), 170.36 KiB | 0 bytes/s, done.
Resolving deltas: 100% (86/86), done.
Checking connectivity... done.
root@kali:# ls -l
total 36
drwxr-xr-x 2 root root 4096 Nov 27 13:31 Desktop
drwxr-xr-x 2 root root 4096 Nov 27 13:31 Documents
drwxr-xr-x 2 root root 4096 Nov 27 17:11 Downloads
drwxr-xr-x 4 root root 4096 Nov 27 17:11 kali-anonsurf
drwxr-xr-x 2 root root 4096 Nov 27 13:31 Music
drwxr-xr-x 2 root root 4096 Nov 27 13:31 Pictures
drwxr-xr-x 2 root root 4096 Nov 27 13:31 Public
drwxr-xr-x 2 root root 4096 Nov 27 13:31 Templates
drwxr-xr-x 2 root root 4096 Nov 27 13:31 Videos
root@kali:# cd kali-anonsurf
root@kali:/kali-anonsurf# ls -l
total 44
-rwxr-xr-x 1 root root 1060 Nov 27 17:11 installer.sh
drwxr-xr-x 5 root root 4096 Nov 27 17:11 kali-anonsurf-deb-src
-rw-r--r-- 1 root root 32472 Nov 27 17:11 LICENSE
-rw-r--r-- 1 root root 1938 Nov 27 17:11 README.md
root@kali:~/kali-anonsurf# sudo ./installer.sh
--2016-11-27 17:12:16-- https://geti2p.net/_static/i2p-debian-repo.key.asc
Resolving geti2p.net (geti2p.net)... 91.143.92.136, 2a02:180🅰️65:2456:6542:1101:1010
Connecting to geti2p.net (geti2p.net)|91.143.92.136|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 14455 (14K) [text/plain]
Saving to: ‘/tmp/i2p-debian-repo.key.asc’

/tmp/i2p-debian-rep 100%[===================>] 14.12K 46.3KB/s in 0.3s

2016-11-27 17:12:18 (46.3 KB/s) - ‘/tmp/i2p-debian-repo.key.asc’ saved [14455/14455]

OK
Hit:2 http://deb.i2p2.no unstable InRelease
Hit:1 http://kali.mirror.garr.it/mirrors/kali kali-rolling InRelease
Reading package lists... Done
Reading package lists... Done
Building dependency tree
Reading state information... Done
i2p-keyring is already the newest version (2016.10.01).
You might want to run 'apt-get -f install' to correct these:
The following packages have unmet dependencies:
kali-anonsurf : Depends: i2p but it is not going to be installed
Recommends: gtkdialog but it is not installable
Recommends: gufw but it is not going to be installed
Recommends: libnotify-bin but it is not going to be installed
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
Reading package lists... Done
Building dependency tree
Reading state information... Done
secure-delete is already the newest version (3.1-6).
secure-delete set to manually installed.
tor is already the newest version (0.2.8.9-1).
tor set to manually installed.
You might want to run 'apt-get -f install' to correct these:
The following packages have unmet dependencies:
i2p : Depends: i2p-router (>= 0.8.6-5) but it is not going to be installed
Depends: libjbigi-jni but it is not going to be installed
Depends: service-wrapper but it is not going to be installed
E: Unmet dependencies. Try 'apt-get -f install' with no packages (or specify a solution).
dpkg-deb: building package 'kali-anonsurf' in 'kali-anonsurf.deb'.
(Reading database ... 303660 files and directories currently installed.)
Preparing to unpack kali-anonsurf.deb ...
Unpacking kali-anonsurf (1.2.2) over (1.2.2) ...
dpkg: dependency problems prevent configuration of kali-anonsurf:
kali-anonsurf depends on i2p; however:
Package i2p is not installed.

dpkg: error processing package kali-anonsurf (--install):
dependency problems - leaving unconfigured
Processing triggers for systemd (231-4) ...
Errors were encountered while processing:
kali-anonsurf
Reading package lists... Done
Building dependency tree
Reading state information... Done
Correcting dependencies... Done
The following packages were automatically installed and are no longer required:
libservice-wrapper-java libservice-wrapper-jni secure-delete service-wrapper
tor tor-geoipdb torsocks
Use 'sudo apt autoremove' to remove them.
The following packages will be REMOVED:
kali-anonsurf
0 upgraded, 0 newly installed, 1 to remove and 1326 not upgraded.
1 not fully installed or removed.
After this operation, 65.5 kB disk space will be freed.
Do you want to continue? [Y/n] Y
(Reading database ... 303660 files and directories currently installed.)
Removing kali-anonsurf (1.2.2) ...
Processing triggers for systemd (231-4) ...
Selecting previously unselected package kali-anonsurf.
(Reading database ... 303652 files and directories currently installed.)
Preparing to unpack kali-anonsurf.deb ...
Unpacking kali-anonsurf (1.2.2) ...
dpkg: dependency problems prevent configuration of kali-anonsurf:
kali-anonsurf depends on i2p; however:
Package i2p is not installed.

dpkg: error processing package kali-anonsurf (--install):
dependency problems - leaving unconfigured
Processing triggers for systemd (231-4) ...
Errors were encountered while processing:
kali-anonsurf
root@kali:~/kali-anonsurf# anonsurf

Parrot AnonSurf Module
Usage:
┌──[root@kali]─[/root/kali-anonsurf]
└──╼ $ anonsurf {start|stop|restart|change|status}

 start - Start system-wide anonymous
	  tunneling under TOR proxy through iptables	  
 stop - Reset original iptables settings
	  and return to clear navigation
 restart - Combines "stop" and "start" options
 change - Changes identity restarting TOR
 status - Check if AnonSurf is working properly
 myip - Show your current IP address
----[ I2P related features ]----
 starti2p - Start i2p services
 stopi2p - Stop i2p services
root@kali:~/kali-anonsurf# anonsurf start

killing dangerous applications
cleaning some dangerous cache elements
[ i ] Stopping IPv6 services:
[ i ] Starting anonymous mode:

Tor is not running! starting it for you

Saved iptables rules

Modified resolv.conf to use Tor and Private Internet Access DNS

All traffic was redirected throught Tor

[ i ] You are under AnonSurf tunnel

root@kali:~/kali-anonsurf# anonsurf myip

My ip is:

curl: (6) Could not resolve host: myexternalip.com

root@kali:~/kali-anonsurf#

**Comment by Und3rf10w** [comment]
What's the output of ```dpkg -l |grep kali-anonsurf```?

If the version number is 1.2.2 and not 1.2.2.1, please pull down the latest version. Based on the dates that are in what you pasted, it appears you're using an old version.

**Comment by Und3rf10w** [comment]
Duplicate of #17

**Comment by Und3rf10w** [comment]
User reported this was solved in #13. I'm going to go ahead and close this.

**Comment by blackghostt** [comment]
cannot resolve this problems

**Comment by ycyashchavan0** [comment]
curl: (6) Could not resolve host: myexternalip.com

**Comment by amreos** [comment]
Same here
this is driving me crazy, it use to work but now neither tor nor anonsurf work
tor browser work when i use openvpn
but anonsurf is dead

**Comment by kipling87** [comment]
Hi all, 
I'm new to github and Kali in general, so let me off easy if I'm being a newb. I have the same problem with anonsurf and tor in general. I am running Kali fully updated (as of today) on a persistent USB. I have two wifi adapters, one being the onboard wifi card on my laptop which is a Intel(R) Dual Band Wireless-AC 7265. The second is a Panda Wireless PAUO5, now when I use kali on the persistent USB it looks like it merges the two connections into one WiFi adapter. (don't know if this is important, just trying to give as much info as possible). Running on the persistent USB I cannot connect to the internet once anonsurf is running I also can't see any results for the anonsurf myip command. However, if I run kali on virtualbox anonsurf works just fine. Also important to note is that I primarily connect through boingo wifi hotspots at the moment. I did think that maybe boingo blocked tor but that doesn't seem to be the case. Any help in resolving this issue would be greatly appreciated, also I  do not have UFW running or anything like that.

**Comment by Herobrine122100** [comment]
hey i was just wondering if there was a way to use anonsurf with a windows computer using the git hub terminal?

**Comment by osxdos** [comment]
thats the most ridiculous shit i ever heard

**Comment by curtisie** [comment]
> curl: (6) Could not resolve host: myexternalip.com

am facing the same problem please help
